python/ql/src/semmle/python/dataflow/new/TypeTracker.qll

```diff
@@ -324,3 +324,144 @@ module TypeTracker {
    */
   TypeTracker end() { result.end() }
 }
+
+private newtype TTypeBackTracker = MkTypeBackTracker(Boolean hasReturn, OptionalAttributeName attr)
+
+/**
+ * Summary of the steps needed to back-track a use of a value to a given dataflow node.
+ *
+ * This can be used to track callbacks that are passed to a certian API call, and are
+ * therefore expected to called with a certain type of value.
```

Comment:
I know this is probably just a copy from JS, but it took me a bit of time to understand it to be honest :smile: 

```suggestion
 * This can for example be used to track callbacks that are passed to a certain API,
 * so we can model specific parameters of that callback as having a certain type.
```

Reply:
If you for some reason would rather keep it intact, here is a bit of grammar cleanup

```suggestion
 * This can be used to track callbacks that are passed to a certain API call, and are
 * therefore expected to be called with a certain type of value.
```

Reply:
I like your first suggestion. :+1:
Also I should add a change note.

Reply:
While I certainly _can_ commit my own suggestion, I would feel more comfortable if you did that :stuck_out_tongue:

Reply:
Did I indicate that I wasn't going to? :confused: 
I just got distracted by aibaars' Ruby puzzle. :smile:

Reply:
> Did I indicate that I wasn't going to? :confused:

The fact that you liked it, but didn't press the button to commit it confused me :confused: